modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java

```diff
@@ -122,20 +125,67 @@ public String getTargetLanguageId() {
 		};
 	}
 
+	private String _getLanguageCode(boolean targetLanguage, String languageId) {
+		List<String> parts = com.liferay.petra.string.StringUtil.split(
+			languageId, CharPool.UNDERLINE);
+
+		String firstPart = parts.get(0);
+
+		// AWS, Azure, DeepL, and GoogleCloud all expect ISO-639 language codes.
+		// ISO-639:1989 renamed "in" to "id." See LPD-23561.
+
+		if (firstPart.equals("in")) {
+			return "id";
+		}
+
+		String secondPart = parts.get(1);
+
+		if (secondPart.equals("ES")) {
+			return "es";
+		}
+
+		StringBundler sb = new StringBundler(5);
+
+		sb.append(firstPart);
+
+		if (targetLanguage) {
+			if (firstPart.equals("en") || firstPart.equals("pt")) {
+				sb.append(CharPool.DASH);
+				sb.append(secondPart);
+			}
+
+			if (firstPart.equals("zh")) {
```

Comment:
@gergelyszalay, I would just add this condition as an "else" instead of another if

I was thinking of an early return with if(!targetLanguage) but I think we should be good to go as is.

My only concern in general would be if getLanguage from the base clase is modified

Reply:
Hi @jkappler, thank you for reviewing, in some cases i dont need to append the second part. like langugae code: "hu". I could group up the zh with en and pt in the same if?